compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/diagnostic/BLangDiagnosticLog.java

```diff
@@ -240,6 +241,14 @@ private void reportDiagnostic(PackageID packageID, DiagnosticCode diagnosticCode
             return;
         }
 
+        for (Object arg : args) {
+            if (arg instanceof BType) {
+                if ((((BType) arg).getKind() == TypeKind.OTHER) && errorCount > 1) {
```

Comment:
Shall we move the `errorCount > 1` check out of the for loop? Since it won't change here no need to check in every loop